src/main/java/net/neoforged/neoforge/fluids/crafting/SimpleFluidIngredient.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids.crafting;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.DynamicOps;
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.MapLike;
+import com.mojang.serialization.RecordBuilder;
+import java.util.stream.Stream;
+import net.minecraft.core.Holder;
+import net.minecraft.core.HolderSet;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.HolderSetCodec;
+import net.minecraft.util.ExtraCodecs;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.Fluids;
+import net.neoforged.neoforge.common.NeoForgeMod;
+import net.neoforged.neoforge.fluids.FluidStack;
+
+/**
+ * Fluid ingredient that matches the fluids specified by the given {@link HolderSet}.
+ * Most commonly, this will either be a list of fluids or a fluid tag.
+ * <p>
+ * Unlike with ingredients, this is technically an explicit "type" of fluid ingredient,
+ * though in JSON, it is still written <b>without</b> a type field, see {@link FluidIngredientCodecs#codec()}
+ */
+public class SimpleFluidIngredient extends FluidIngredient {
+    public static final Codec<HolderSet<Fluid>> HOLDER_SET_NO_EMPTY_FLUID = HolderSetCodec.create(
+            Registries.FLUID, FluidStack.FLUID_NON_EMPTY_CODEC, false);
+
+    public static final Codec<SimpleFluidIngredient> CODEC = ExtraCodecs.nonEmptyHolderSet(HOLDER_SET_NO_EMPTY_FLUID)
+            .xmap(SimpleFluidIngredient::new, SimpleFluidIngredient::fluidSet);
+
+    // Note: This map codec explicitly always errors, since we want to force people to use HolderSet syntax!
+    public static final MapCodec<SimpleFluidIngredient> MAP_CODEC = new MapCodec<>() {
```

Comment:
Doesn't need to be exposed if you can avoid it.

Reply:
Unfortunately I can't unless I want to specify the IngredientType in this file